### PR TITLE
ROE-2414 BO Gov id changes

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-2414

### Change description

* Changes to allow addition, modification and removal of BO Gov entities with ids being present in the URLs
* New tests added (duplicated)

### Work checklist

- [X] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/beneficial.owner.gov.ts
Comment (mwestacott) [inline src/utils/beneficial.owner.gov.ts:65]: Need to refactor this, once @markpit 's PR is merged. - https://github.com/companieshouse/overseas-entities-web/pull/1143.

File: src/controllers/beneficial.owner.type.controller.ts
Comment (mwestacott) [reply src/controllers/beneficial.owner.type.controller.ts:89]: 👍 yes, seen that. I'll wait for @mattch1 's branch to be merged in and then refactor.

File: src/controllers/beneficial.owner.gov.controller.ts
Comment (dlloyd1-ch) [inline src/controllers/beneficial.owner.gov.controller.ts:16]: Just wondering how this works without referencing BENEFICIAL_OWNER_TYPE_WITH_PARAMS_URL? Same for  getById in line 20.

File: src/utils/beneficial.owner.gov.ts
Comment (mwestacott) [reply src/utils/beneficial.owner.gov.ts]: Done.

File: src/controllers/beneficial.owner.type.controller.ts
Comment (mwestacott) [reply src/controllers/beneficial.owner.type.controller.ts]: Refactoring complete.

File: src/controllers/beneficial.owner.gov.controller.ts
Comment (mwestacott) [reply src/controllers/beneficial.owner.gov.controller.ts]: Well it works/worked because the fourth parameter being passed in (originally `config.BENEFICIAL_OWNER_TYPE_URL`) was used to set the 'Back' link and as that is another story I didn't bother changing those at this point.

However, I saw that @mattch1 did in his implementation for BO other and as it was such a simple change and to remain consistent I've now changed this here too.

**Review by dlloyd1-ch — APPROVED** [review]

File: src/controllers/beneficial.owner.gov.controller.ts
Comment (dlloyd1-ch) [reply src/controllers/beneficial.owner.gov.controller.ts]: Ah, ok, if its back link related we probably should leave it until the back link story but no point reverting now for such a small change.

**Pull request merged**
